### PR TITLE
Update to Processing 4.2

### LINKED ISSUE #1111: Upgrade to Processing 4
## Problem

Processing 3 is no longer supported and several workarounds are needed in order to continue supporting modern systems. We should upgrade to Processing 4 or enumerate any obstacles that prevent us from doing so.

## Expected 

The application will work when run from Processing 4.

### LINKED ISSUE #1162: Add MacOS builds to Github Actions instead of Travis
## **Is your feature request related to a problem? Please describe.**
We have automated builds for Windows and Linux distributions, but not for MacOS. This is mainly due to application signing requirements enforced by Apple which have stopped previous attempts to automate the builds.

## **Describe the solution you'd like**
Automatically build MacOS releases as part of the GitHub Actions workflow.

## **Describe alternatives you've considered**
In order to sign applications for MacOS, the signing key must be added to the system keychain. This requires administrative privileges and is blocked on GitHub Actions work nodes. It may be possible to build the application in a Docker container to resolve this administrative access issues.

## **Additional context**
https://github.com/sickcodes/Docker-OSX

### LINKED ISSUE #1170: Add Windows builds via Github Actions
As part of Processing 4 update, migrate to building GUI for CI/CD using Github Actions.

### LINKED ISSUE #1171: Add Linux builds via Github Actions
As part of Processing 4 update, let's build the GUI for CI/CD using Github Actions.

### PR DESCRIPTION
**Pull request by retiutut**
**Modify App to be fully compatible with Processing 4**

Closes #1111 

There are still issues with child windows not being able to be resized and closed properly. To help us move forward for now, FilterUI, ConsoleLog, and EMG Settings UI have been made non-resizeable. A boolean has been left in FilterUI that would allow this again and easy testing, should a fix become available.

___

**CI/CD Update**

Closes #1170 
Closes #1162 
Closes #1171

### CODE REVIEW

**Review by Rexblk81 — APPROVED** [review]

File: release_script/mac_only/Certificates_2023.p12.enc
Comment (philippitts) [inline release_script/mac_only/Certificates_2023.p12.enc:1]: Can this be made into a build variable so it's not exposed as a file?

File: release_script/mac_only/dmgbuild_settings_old.py
Comment (philippitts) [inline release_script/mac_only/dmgbuild_settings_old.py:1]: Unless this is actively used now, it should be removed. The git history will keep track of old files.

**Review by philippitts — CHANGES_REQUESTED** [review]
- [x] Recommend removing the debug lines that show up in the console log whenever the window is moved.
- [x] There is no window resizing cursor on Windows. Can this be added?

File: release_script/mac_only/Certificates_2023.p12.enc
Comment (retiutut) [reply release_script/mac_only/Certificates_2023.p12.enc:1]: It cannot.

Comment (philippitts) [reply release_script/mac_only/Certificates_2023.p12.enc:1]: https://stackoverflow.com/questions/70791640/how-to-set-gihub-secret-with-a-binary-file-for-github-workflow

**Comment by retiutut** [comment]
> [ ] Recommend removing the debug lines that show up in the console log whenever the window is moved.

Steps to replicate?

File: release_script/mac_only/Certificates_2023.p12.enc
Comment (philippitts) [reply release_script/mac_only/Certificates_2023.p12.enc:1]: https://docs.github.com/en/actions/security-guides/encrypted-secrets#storing-base64-binary-blobs-as-secrets

Comment (retiutut) [reply release_script/mac_only/Certificates_2023.p12.enc:1]: This changes the process for CI/CD a good bit I think.

**Comment by philippitts** [comment]
> > [ ] Recommend removing the debug lines that show up in the console log whenever the window is moved.
> 
> Steps to replicate?

Launch from Processing 4 and then move the GUI window around. You should see something like this:

```
__MOVE__ 589 760
__MOVE__ 589 761
__MOVE__ 588 761
__MOVE__ 588 762
__MOVE__ 588 763
__MOVE__ 588 764
__MOVE__ 587 765
__MOVE__ 587 766
__MOVE__ 587 767
__MOVE__ 587 768
__MOVE__ 587 769
__MOVE__ 586 769
```

**Comment by retiutut** [comment]
> > > [ ] Recommend removing the debug lines that show up in the console log whenever the window is moved.
> > 
> > 
> > Steps to replicate?
> 
> Launch from Processing 4 and then move the GUI window around. You should see something like this:
> 
> ```
> __MOVE__ 589 760
> __MOVE__ 589 761
> __MOVE__ 588 761
> __MOVE__ 588 762
> __MOVE__ 588 763
> __MOVE__ 588 764
> __MOVE__ 587 765
> __MOVE__ 587 766
> __MOVE__ 587 767
> __MOVE__ 587 768
> __MOVE__ 587 769
> __MOVE__ 586 769
> ```

This is something added by Processing 4. Only shows when running from the Processing 4 IDE.

>There is no window resizing cursor on Windows. Can this be added?

We can resize windows from the top left.

File: release_script/mac_only/Certificates_2023.p12.enc
Comment (retiutut) [reply release_script/mac_only/Certificates_2023.p12.enc:1]: Ok will do!

**Review by philippitts — APPROVED** [review]

**Pull request merged**